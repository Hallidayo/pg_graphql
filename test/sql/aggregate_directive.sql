begin;

-- Create a simple table without any directives
create table product(
    id serial primary key,
    name text not null,
    price numeric not null,
    stock int not null
);

insert into product(name, price, stock)
values
    ('Widget', 9.99, 100),
    ('Gadget', 19.99, 50),
    ('Gizmo', 29.99, 25);

-- Try to query aggregate without enabling the directive - should fail
select graphql.resolve($$
{
  productCollection {
    aggregate {
      count
    }
  }
}
$$);

-- Enable aggregates
comment on table product is e'@graphql({"aggregate": {"enabled": true}})';

-- Now aggregates should be available - should succeed
select graphql.resolve($$
{
  productCollection {
    aggregate {
      count
      sum {
        price
        stock
      }
      avg {
        price
      }
      max {
        price
        name
      }
      min {
        stock
      }
    }
  }
}
$$);

rollback; 