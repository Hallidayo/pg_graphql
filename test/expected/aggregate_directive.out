begin;

-- Create a simple table without any directives
create table product(
    id serial primary key,
    name text not null,
    price numeric not null,
    stock int not null
);

insert into product(name, price, stock)
values
    ('Widget', 9.99, 100),
    ('Gadget', 19.99, 50),
    ('Gizmo', 29.99, 25);

-- Try to query aggregate without enabling the directive - should fail
select graphql.resolve($$
{
  productCollection {
    aggregate {
      count
    }
  }
}
$$);
                                resolve                                 
------------------------------------------------------------------------
 {"data": null, "errors": [{"message": "unknown field in connection"}]}
(1 row)

-- Enable aggregates
comment on table product is e'@graphql({"aggregate": {"enabled": true}})';

-- Now aggregates should be available - should succeed
select graphql.resolve($$
{
  productCollection {
    aggregate {
      count
      sum {
        price
        stock
      }
      avg {
        price
      }
      max {
        price
        name
      }
      min {
        stock
      }
    }
  }
}
$$);
                                                                           resolve                                                                            
--------------------------------------------------------------------------------------------------------------------------------------------------------------
 {"data": {"productCollection": {"aggregate": {"avg": {"price": 19.99}, "count": 3, "max": {"name": "Widget", "price": 29.99}, "min": {"stock": 25}, "sum": {"price": 59.97, "stock": 175}}}}}
(1 row)

rollback; 